packages/engine/src/scene/components/MixerComponent.tsx

```diff
@@ -0,0 +1,367 @@
+  [MixableType.Number]: {
+    create: (a) => a ?? 0,
+    lerp: (a: number, b: number, p) => a * (1 - p) + b * p,
+    fromNumberList: (a: number[]) => a[0],
+    toNumberList: (a: number) => [a]
+  },
+  [MixableType.Vector2]: {
+    create: (a?: number[]) => (a == null ? vec2.create() : vec2.fromValues(...(a as [number, number]))),
+    lerp: (a: vec2, b: vec2, p) => vec2.lerp(vec2.create(), a, b, p),
+    fromNumberList: (a: number[]) => vec2.fromValues(...(a as [number, number])),
+    toNumberList: (a: vec2) => [...a]
+  },
+  [MixableType.Vector3]: {
+    create: (a?: number[]) => (a == null ? vec3.create() : vec3.fromValues(...(a as [number, number, number]))),
+    lerp: (a: vec3, b: vec3, p) => vec3.lerp(vec3.create(), a, b, p),
+    fromNumberList: (a: number[]) => vec3.fromValues(...(a as [number, number, number])),
+    toNumberList: (a: vec3) => [...a]
+  },
+  [MixableType.Vector4]: {
+    create: (a?: number[]) => (a == null ? vec4.create() : vec4.fromValues(...(a as [number, number, number, number]))),
+    lerp: (a: vec4, b: vec4, p) => vec4.lerp(vec4.create(), a, b, p),
+    fromNumberList: (a: number[]) => vec4.fromValues(...(a as [number, number, number, number])),
+    toNumberList: (a: vec4) => [...a]
+  },
+  [MixableType.Quaternion]: {
+    create: (a?: number[]) => (a == null ? quat.create() : quat.fromValues(...(a as [number, number, number, number]))),
+    lerp: (a: quat, b: quat, p) => quat.lerp(quat.create(), a, b, p),
+    fromNumberList: (a: number[]) => quat.fromValues(...(a as [number, number, number, number])),
+    toNumberList: (a: quat) => [...a]
+  }
+}
+
+type AnyComponent = Component<any, any, any, any, any, any>
+type AnyComponentWithID = AnyComponent & { jsonID: string }
+
+const toEntityUUID = (entity: Entity | EntityUUID): EntityUUID =>
+  typeof entity === 'string' ? entity : getComponent(entity, UUIDComponent)
+const toEntity = (entity: Entity | EntityUUID): Entity =>
+  typeof entity === 'string' ? UUIDComponent.getEntityByUUID(entity) : entity
+
+const toComponentID = (targetComponent: AnyComponentWithID | string): string | undefined =>
+  typeof targetComponent === 'string' ? targetComponent : targetComponent.jsonID
+const toComponent = (targetComponent: AnyComponentWithID | string): AnyComponent | undefined =>
+  typeof targetComponent === 'string' ? ComponentJSONIDMap.get(targetComponent) : targetComponent
+
+type PropertyAddress = [EntityUUID, string, string]
+
+const packAddress = (
+  targetEntity: Entity | EntityUUID,
+  targetComponent: AnyComponentWithID | string,
+  propertyPath: string
+): string => `${toEntityUUID(targetEntity)}::${toComponentID(targetComponent)}::${propertyPath}`
+
+const unpackAddress = (packedAddress: string): PropertyAddress =>
+  packedAddress.split('::') as [EntityUUID, string, string]
+
+type Entry = Record<string, number[]>
+type Property = { type: MixableType; address: PropertyAddress }
+
+const createProperty = (
```

Comment:
@dinomut1 Two important things not yet implemented in the component's properties are support for properties nestled in their schema, and deriving the type of a property from the schema.

Reply:
Now implemented. 👍